client.go

```diff
@@ -427,6 +430,20 @@ func (c *Client) stream(method, path string, streamOptions streamOptions) error
 		if err != nil {
 			return err
 		}
+
+		// ReadResponse may hang if server does not replay
+		connected := make(chan bool, 2)
+		if streamOptions.timeout > 0 {
+			go func() {
+				select {
+				case <-time.After(streamOptions.timeout):
```

Comment:
Why not having just the receive and the dial.Close? Just like below:

```
<-time.After(streamOptions.timeout)
dial.Close()
```

You have a receive operation on connected, but no sends nor closes, so I can't really tell why connected exists.